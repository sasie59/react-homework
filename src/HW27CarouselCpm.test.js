import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import HW from'./HW27CarouselCpm';

jest.useFakeTimers();

test('should HW27CarouselCpm', () => {
  act(() => {
    const {container } = render(
      <HW
        time={1500}
        isMoving={true}
      >
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </HW>
    );
    jest.advanceTimersByTime(1500);
    expect(container).toMatchSnapshot();
    jest.advanceTimersByTime(1500);
    expect(container).toMatchSnapshot();
    jest.advanceTimersByTime(1500);
    expect(container).toMatchSnapshot();
  });
});
