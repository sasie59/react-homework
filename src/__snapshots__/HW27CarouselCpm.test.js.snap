// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW27CarouselCpm 1`] = `
<div>
  <div>
    <div
      class="carousel"
    >
      <div
        class="item hide "
      >
        <div>
          1
        </div>
      </div>
      <div
        class="item show "
      >
        <div>
          2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should HW27CarouselCpm 2`] = `
<div>
  <div>
    <div
      class="carousel"
    >
      <div
        class="item "
      >
        <div>
          2
        </div>
      </div>
      <div
        class="item "
      >
        <div>
          3
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should HW27CarouselCpm 3`] = `
<div>
  <div>
    <div
      class="carousel"
    >
      <div
        class="item "
      >
        <div>
          3
        </div>
      </div>
      <div
        class="item "
      >
        <div>
          1
        </div>
      </div>
    </div>
  </div>
</div>
`;
